Update main.nf

fhartley101 opened the pull request:
Lots of changes:
1. Update module names in info
2. Added the instructions for cutadapt_options where user inputs their chosen options for the cutadapt function in a txt file
3. Added a section in the log info to read out the cutadapt options file
4. Import both new modules (FASTQC_CUTADAPT and CUTADAPT)
5. Added the cutadapt module The first section deals with fastqc_cutadapt being passed without cutadapt.	fastqc_cutadapt and fastqc can't be run together if cutadapt is not also specified (they'd both received the same input and thus duplicate) so this produces an error message. fastqc_cutadapt will work if cutadapt isn't specified, but the output of cutadapt (i.e. the trimmed files) is supplied in input. The directory MUST be named *cutadapt/ otherwise it will error The second section deals with cutadapt being run. It imports the cutadapt options from the user provided txt file then runs cutadapt. It then asks if fastqc_cutadapt is also specified in modules_to_run and runs that if needed.
6. Removes the ch_salmon_multiqc section as this is not necessary now multiqc reads from the output directory. If salmon didn't run, there would be an empty channel which means multiqc couldn't be run on fastqc results alone in the original pipeline.
7. Remove the extra inputs to multiqc as they are no longer needed
8. Added a section to export the multiqc version used

fhartley101 committed:
Update main.nf

Lots of changes:
1. Update module names in info
2. Added the instructions for cutadapt_options where user inputs their chosen options for the cutadapt function in a txt file
3. Added a section in the log info to read out the cutadapt options file
4. Import both new modules (FASTQC_CUTADAPT and CUTADAPT)
5. Added the cutadapt module
The first section deals with fastqc_cutadapt being passed without cutadapt.	fastqc_cutadapt and fastqc can't be run together if cutadapt is not also specified (they'd both received the same input and thus duplicate) so this produces an error message.
fastqc_cutadapt will work if cutadapt isn't specified, but the output of cutadapt (i.e. the trimmed files) is supplied in input. The directory MUST be named *cutadapt/ otherwise it will error
The second section deals with cutadapt being run. It imports the cutadapt options from the user provided txt file then runs cutadapt. It then asks if fastqc_cutadapt is also specified in modules_to_run and runs that if needed.
6. Removes the ch_salmon_multiqc section as this is not necessary now multiqc reads from the output directory. If salmon didn't run, there would be an empty channel which means multiqc couldn't be run on fastqc results alone in the original pipeline.
7. Remove the extra inputs to multiqc as they are no longer needed
8. Added a section to export the multiqc version used

fhartley101 committed:
Update main.nf

This version uses the existing fastqc module to run fastqc on the cutadapt output. A new channel is added so that a different output directory can be specified for when the module runs before and after cutadapt

alebarberis commented on main.nf:
```diff
@@ -244,6 +253,47 @@ workflow {
+    //---------------------------------------------
+    // Adapter trimming
+    //---------------------------------------------    
+    if(modules_to_run.contains('fastqc_cutadapt') && !modules_to_run.contains('cutadapt')) {
+        if(modules_to_run.contains('fastqc')){
+            exit 1, 'ERROR: The fastqc and fastqc_cutadapt modules may not be specified together unless the cutadapt module is also present' 
+        }
+        if(!modules_to_run.contains('fastqc') && params.input.endsWith('cutadapt/')){
+            ch_fastqc_dir = 'fastqc_cutadapt'
+            FASTQC_CUTADAPT(
+                ch_reads,
+                ch_fastqc_dir
+            )
+            // Update versions
+            ch_versions = ch_versions.mix(FASTQC_CUTADAPT.out.versions.first())
+        } else {
+            exit 1, 'ERROR: The fastqc_cutadapt module is specified without the cutadapt module, and the input files are not stored in a cutadapt/ directory. Update the input directory and the suffix parameters to feed the trimmed files into fastqc_cutadapt directly.'
+        }
+    }
```
Check if this can be merged with the section below.

fhartley101 committed:
Update main.nf

Changed logic for cutadapt module as requested

fhartley101 committed:
Update main.nf

Added the lines to read in the cutadapt options file as that matches the current github versions
(I have since removed this but will put those updates on a separate branch)

fhartley101 committed:
Update main.nf

Changes to cutadapt logic again to remove duplication of module call
